spharpy/spherical.py

```diff
@@ -1115,3 +1118,385 @@ def sph_identity_matrix(n_max, type='n-nm'):
+        """Get or set the gradient of the basis
+        matrix with respect to phi."""
+        if self._basis_gradient_phi is None:
+            self._compute_basis_gradient()
+        return self._basis_gradient_phi
+
+    def _compute_basis(self):
+        """
+        Compute the basis matrix for the SphericalHarmonics class.
+        """
+        if self.basis_type == "complex":
+            function = spherical_harmonic_basis
+        elif self.basis_type == "real":
+            function = spherical_harmonic_basis_real
+        else:
+            raise ValueError(
+                "Invalid basis type, should be either 'complex' or 'real'"
+            )
+        self._basis = function(
+            self.n_max, self.coordinates,
+            self.normalization, self.channel_convention)
+
+    def _compute_basis_gradient(self):
+        """
+        Compute the gradient of the basis matrix for the SphericalHarmonics class
+        """
+        if any(
+            (self.normalization in ["maxN", "sn3d"],
+             self.channel_convention == "fuma")
+        ):
+            raise ValueError(
+            f"Gradient computation not supported for normalization "
+            f"'{self.normalization}' and "
+            f"channel convention '{self.channel_convention}'."
+            )
+        else:
+            if self.basis_type == "complex":
+                function = spherical_harmonic_basis_gradient
+            elif self.basis_type == "real":
+                function = spherical_harmonic_basis_gradient_real
+            else:
+                raise ValueError(
+                    "Invalid basis type, should be either 'complex' or 'real'"
+                )
+            self._basis_gradient_theta, self._basis_gradient_phi = function(
+                self.n_max, self.coordinates)
+
+    @property
+    def basis_inv(self):
+        """Get or set the inverse basis matrix."""
+        if self._basis is None:
+            self._compute_basis()
+        if self._basis_inv is None:
+            self._compute_inverse()
+        return self._basis_inv
+
+    def _compute_inverse(self):
+        """
+        Compute the inverse basis matrix for the SphericalHarmonics class
+        """
+        if self._basis is None:
+            self._compute_basis()
+        if not isinstance(self.coordinates, SamplingSphere):
+            warnings.warn("Using a simple pf.Coordinates object for inverse_basis." \
+            " Make sure this is intended.")
+        if self.inverse_method == "pseudo_inverse":
+            self._basis_inv = np.linalg.pinv(self._basis)
+        elif self.inverse_method == "quadrature":
+            self._basis_inv = np.einsum('ij,i->ji', np.conj(self._basis), 
+                                        self.coordinates.weights)
+        else:
+            raise ValueError(
+                "Invalid inverse transform method. "
+                "Allowed values are 'pseudo_inverse', 'quadrature', or 'auto'."
+            )
```

Comment:
I think this case should be deleted, since the value of `inverse_method` is checked in its setter.